### PR TITLE
Improve link ordering, highlighting

### PR DESCRIPTION
**Pull request by jacobp100**
https://ag-grid.atlassian.net/browse/AG-9528
https://ag-grid.atlassian.net/browse/AG-11561

A lot of duplication again - I want to get labels and some other features in before refactoring

### CODE REVIEW

File: packages/ag-charts-community/src/chart/series/chord/chordSeries.ts [hunk 1]
```diff
@@ -351,25 +391,102 @@ export class ChordSeries extends DataModelSeries<
...
+                datumSelection: this.highlightNodeSelection,
+            });
+            await this.updateNodeNodes({ datumSelection: this.highlightNodeSelection, isHighlight: true });
+
+            this.highlightLinkSelection = await this.updateLinkSelection({
+                nodeData: [],
+                datumSelection: this.highlightLinkSelection,
+            });
+            await this.updateLinkNodes({ datumSelection: this.highlightLinkSelection, isHighlight: true });
+        }
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/series/chord/chordSeries.ts]: Any way we can improve the signal to noise ratio in terms of readability here, it looks like there are 4 `this.updateNodeSelection()` calls happening on all 3 branches, and only some of the parameters are changed?

File: packages/ag-charts-community/src/chart/series/sankey/sankeySeries.ts [hunk 2]
```diff
@@ -390,25 +411,102 @@ export class SankeySeries
...
-
         this.linkSelection = await this.updateLinkSelection({ nodeData, datumSelection: this.linkSelection });
         await this.updateLinkNodes({ datumSelection: this.linkSelection, isHighlight: false });
 
-        this.highlightLinkSelection = await this.updateLinkSelection({
-            nodeData: highlightedDatum?.type === SankeyDatumType.Link ? [highlightedDatum] : [],
-            datumSelection: this.highlightLinkSelection,
-        });
-        await this.updateLinkNodes({ datumSelection: this.highlightLinkSelection, isHighlight: true });
+        if (highlightedDatum?.type === SankeyDatumType.Node) {
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/series/sankey/sankeySeries.ts:409]: Ditto

File: packages/ag-charts-community/src/chart/series/chord/chordSeries.ts (hunk 1 above)
Comment (jacobp100) [reply packages/ag-charts-community/src/chart/series/chord/chordSeries.ts]: Updated

File: packages/ag-charts-community/src/chart/series/sankey/sankeySeries.ts (hunk 2 above)
Comment (jacobp100) [reply packages/ag-charts-community/src/chart/series/sankey/sankeySeries.ts:409]: Updated

**Review by alantreadway — APPROVED** [review]
LGTM, thanks for the updates!

**Pull request merged**